dell/src/main/java/org/apache/iceberg/dell/ecs/EcsCatalog.java

```diff
@@ -0,0 +1,517 @@
+        path != null && path.length() > 0,
+        "Cannot initialize EcsCatalog because warehousePath must not be null");
+    int len = path.length();
+    if (path.endsWith(delimiter)) {
+      return path.substring(0, len - delimiter.length());
+    } else {
+      return path;
+    }
+  }
+
+  private FileIO initializeFileIO(Map<String, String> properties) {
+    String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
+    if (fileIOImpl == null) {
+      FileIO io = new EcsFileIO();
+      io.initialize(properties);
+      return io;
+    } else {
+      return CatalogUtil.loadFileIO(fileIOImpl, properties, hadoopConf);
+    }
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    return new EcsTableOperations(catalogName + "." + tableIdentifier,
+        tableURI(tableIdentifier), fileIO, this);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    StringBuilder builder = new StringBuilder();
+    builder.append(warehouseLocation);
+    for (String level : tableIdentifier.namespace().levels()) {
+      builder.append(dellProperties.ecsCatalogDelimiter());
+      builder.append(level);
+    }
+
+    builder.append(dellProperties.ecsCatalogDelimiter());
+    builder.append(tableIdentifier.name());
+    return builder.toString();
+  }
+
+  /**
+   * Iterate all table objects with the namespace prefix.
+   */
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    if (!namespace.isEmpty() && !namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist", namespace);
+    }
+
+    String marker = null;
+    List<TableIdentifier> results = Lists.newArrayList();
+    EcsURI prefix = namespacePrefix(namespace);
+    do {
+      ListObjectsResult listObjectsResult = client.listObjects(
+          new ListObjectsRequest(prefix.bucket())
+              .withDelimiter(dellProperties.ecsCatalogDelimiter())
+              .withPrefix(prefix.name())
+              .withMarker(marker));
+      marker = listObjectsResult.getNextMarker();
+      results.addAll(listObjectsResult.getObjects().stream()
+          .filter(s3Object -> s3Object.getKey().endsWith(TABLE_OBJECT_SUFFIX))
+          .map(object -> parseTableId(namespace, prefix, object))
+          .collect(Collectors.toList()));
+    } while (marker != null);
+
+    LOG.debug("Listing of namespace: {} resulted in the following tables: {}", namespace, results);
+    return results;
+  }
+
+  /**
+   * Get object prefix of namespace.
+   */
+  private EcsURI namespacePrefix(Namespace namespace) {
+    String prefix;
+    if (namespace.isEmpty()) {
+      prefix = dellProperties.ecsCatalogPrefix().name();
+    } else {
+      prefix = dellProperties.ecsCatalogPrefix().name() +
+          String.join(dellProperties.ecsCatalogDelimiter(), namespace.levels()) +
+          dellProperties.ecsCatalogDelimiter();
```

Comment:
Because the delimiter listing is different with fs which allows leading '/'. The leading delimiter should be skipped when the prefix has no value.
For example, the bucket has **t1.table**, **n1.namespace**, **n1/t2.table**, listing the table **t1** of empty namespace should use blank prefix ` `, and listing the table **t2** should use prefix `n1/`